pkg/tsdb/stackdriver/stackdriver.go

```diff
@@ -152,43 +152,67 @@ func (e *StackdriverExecutor) buildQueries(tsdbQuery *tsdb.TsdbQuery) ([]*Stackd
 	durationSeconds := int(endTime.Sub(startTime).Seconds())
 
 	for _, query := range tsdbQuery.Queries {
+		migrateLegacyQueryModel(query)
+		q := GrafanaQuery{}
+		model, _ := query.Model.MarshalJSON()
+		if err := json.Unmarshal(model, &q); err != nil {
+			return nil, fmt.Errorf("could not unmarshal StackdriverQuery json: %w", err)
+		}
 		var target string
-
-		metricType := query.Model.Get("metricType").MustString()
-		filterParts := query.Model.Get("filters").MustArray()
-
 		params := url.Values{}
 		params.Add("interval.startTime", startTime.UTC().Format(time.RFC3339))
 		params.Add("interval.endTime", endTime.UTC().Format(time.RFC3339))
-		params.Add("filter", buildFilterString(metricType, filterParts))
-		params.Add("view", query.Model.Get("view").MustString("FULL"))
-		setAggParams(&params, query, durationSeconds)
+
+		sq := &StackdriverQuery{
+			RefID:    query.RefId,
+			GroupBys: []string{},
+		}
+
+		if q.QueryType == metricQueryType {
+			params.Add("filter", buildFilterString(q.MetricQuery.MetricType, q.MetricQuery.Filters))
+			params.Add("view", q.MetricQuery.View)
+			setAggParams(&params, &q.MetricQuery, durationSeconds, query.IntervalMs)
+			sq.GroupBys = append(sq.GroupBys, q.MetricQuery.GroupBys...)
+			sq.AliasBy = q.MetricQuery.AliasBy
+			sq.ProjectName = q.MetricQuery.ProjectName
+		} else if q.QueryType == sloQueryType {
+			sq.AliasBy = q.SloQuery.AliasBy
+			sq.ProjectName = q.SloQuery.ProjectName
+			sq.Selector = q.SloQuery.SelectorName
+			sq.Service = q.SloQuery.ServiceId
+			sq.Slo = q.SloQuery.SloId
+			params.Add("aggregation.alignmentPeriod", calculateAlignmentPeriod(q.SloQuery.AlignmentPeriod, query.IntervalMs, durationSeconds))
+			params.Add("filter", buildSLOFilterExpression(q.SloQuery))
+			if q.SloQuery.SelectorName == "select_slo_health" {
+				params.Add("aggregation.perSeriesAligner", "ALIGN_MEAN")
+			} else {
+				params.Add("aggregation.perSeriesAligner", "ALIGN_NEXT_OLDER")
+			}
```

Comment:
I assume that `aggregation.crossSeriesReducer` parameter is not meaningful for SLO queries?

Reply:
Correct!

Reply:
Not that important but it would be easier for comparing the two different query types to have a similar to `setAggParams` for SLO queries for setting `aggregation.perSeriesAligner` and `aggregation.alignmentPeriod` parameters.

Reply:
I agree! Broke out agg param building for SLOs into its own func